.github/workflows/testing-and-deployment.yml

```diff
@@ -149,21 +149,22 @@ jobs:
       # see discussion at https://github.com/pyvista/pyvista/issues/2867
       matrix:
         include:
+          # numeric numpy versions are < conditions, e.g. "1.24" means "numpy<1.24"
           - python-version: "3.8"
             vtk-version: "9.0.3" # Requires numpy<1.24
-            numpy-version: "default"
+            numpy-version: "1.24"
           - python-version: "3.9"
             vtk-version: "9.1"
-            numpy-version: "default"
+            numpy-version: "2.0" # matplotlib and numpy incomptability for python 3.9
```

Comment:
This indeed sounds weird. We might want to exclude this combination in our `pyproject.toml` unless it's a huge pain in the ass to specify.

Reply:
I'm not super confident this is true, but our CI couldn't resolve compatible dependencies.  This pin is now for testing numpy 1.26 rather than specifically to avoid incompatibility.

If someone wants to put in the time to figure this out, it makes sense to me to put into `pyproject.toml`, although I'm not sure that python dependency specification allows for conditions on dependency versions.  I suppose we could do it by Python version, but it is a bit of an incomplete situation.  What if it is possible for a user on Python 3.10 to also get the problematic combination of the two packages, for example?

It should be up to matplotlib to handle this in an ideal world, but they run on a (much) narrower support window than we do.